x/fswap/client/testutil/tx.go

```diff
@@ -0,0 +1,379 @@
+package testutil
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/gogo/protobuf/proto"
+
+	"github.com/Finschia/finschia-sdk/client/flags"
+	clitestutil "github.com/Finschia/finschia-sdk/testutil/cli"
+	sdk "github.com/Finschia/finschia-sdk/types"
+	sdkerrors "github.com/Finschia/finschia-sdk/types/errors"
+	"github.com/Finschia/finschia-sdk/x/fswap/client/cli"
+)
+
+func (s *IntegrationTestSuite) TestCmdTxMsgSwap() {
+	val := s.network.Validators[0]
+	clientCtx := val.ClientCtx
+	// avoid printing as yaml from CLI command
+	clientCtx.OutputFormat = jsonOutputFormat
+
+	testCases := []struct {
+		name         string
+		args         []string
+		expectErr    bool
+		expectedCode uint32
+		respType     proto.Message
+	}{
+		{
+			"valid transaction",
+			[]string{
+				val.Address.String(),
+				sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10)).String(),
+				s.toDenom.Base,
+				fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+				fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+				fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10))).String()),
+			},
+			false,
+			0,
+			&sdk.TxResponse{},
+		},
+		{
+			"invalid request (wrong number of args)",
+			[]string{
+				val.Address.String(),
+				sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10)).String(),
+				s.toDenom.Base,
+				"extra",
+				fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+				fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+				fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(1))).String()),
+			},
+			true,
+			0,
+			nil,
+		},
+		{
+			"invalid request (invalid from address)",
+			[]string{
+				"invalidAddress",
+				sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10)).String(),
+				s.toDenom.Base,
+				fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+				fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+				fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(1))).String()),
+			},
+			true,
+			0,
+			nil,
+		},
+		{
+			"invalid request (invalid from coin amount)",
+			[]string{
+				val.Address.String(),
+				"",
+				s.toDenom.Base,
+				fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+				fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+				fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(1))).String()),
+			},
+			true,
+			0,
+			nil,
+		},
+		{
```

Comment:
Is this case relevant to the coverage of x/fswap?

Reply:
I want to simulate a situation that `coin amount` is invalid. Or I change it into `1abc1`?

Reply:
I mean, the case for "not enough fees" seems like redundant.

Reply:
Also for "not enough gas".

Reply:
These two tests are referenced by the bank module tests. I think they also can be deleted, I will modify it.